### PR TITLE
Add visual feedback of layers out of range to tree panel

### PR DESCRIPTION
**Pull request by bentrm**
I would like to add some visual feedback for layers that are scale-based and move out of range of the maps resolution. It may be worthwhile to have in the base library as well.

 This is a first draft which makes sure active layers that become non-visible to resolution constraints are unchecked and vice-versa. I'm unsure if this is the most user friendly approach though.

### CODE REVIEW

**Comment by chrismayer** [comment]
Hi @bentrm,

I like the idea but I think it is not the best solution to uncheck the layer since this could irritate the user. I would suggest to disable the layer node and let the layer untoggled.
Any other opinions?

Cheers,
Chris

**Comment by bartvde** [comment]
I agree @chrismayer 
I think what we had in GeoExt 1 wasn't so bad.

**Comment by bentrm** [comment]
There is no obvious way to disable tree rows in ExtJS 4, so I made changes to add a custom class to rows whose layers are out of range.

File: src/GeoExt/tree/View.js [hunk 1]
```diff
@@ -44,6 +44,10 @@ Ext.define('GeoExt.tree.View', {
         return me.callParent(arguments);
     },
 
+    getRowClass: function(record, rowIndex, rowParams, store) {
```
Comment (chrismayer) [inline src/GeoExt/tree/View.js]: Would you mind adding some little API-Docs?

**Comment by chrismayer** [comment]
Thanks @bentrm for your ongoing effort. You are right there seems to be no build in disable-function for tree nodes, so your CSS approach does the job very well. I added a little line comment. Furthermore would you mind adding little unit tests?  

Two more questions (but not blocking a merge IMHO):
  - Should this behaviour be configurable?
  - Should the layer tree node remain checkable (toggle visibility) when beeing disabled?

Cheers,
Chris

**Comment by marcjansen** [comment]
My 2 cents:

Making the checkbox unusable in case the row is disabled can be as easy as adding some CSS:

```css
.gx-tree-row-disabled input {
    visibility: hidden;
}
```

The configurability is another thing. It may be useful, to have it configurable in JS, but again, some CSS may do the trick if one doesn't want the behaviour intruced here:

```css
.gx-tree-row-disabled span.x-tree-node-text {
    opacity: inherit;
    font-style: inherit;
}
```

I am going merge now (both @bartvde and @chrismayer had already given their consent), so this can go into v2.0.3.

My above comment / hints will be added in a follow up commit, so our users can read about this in the API-docs.

Thanks to everybody involved :+1:

**Pull request merged**